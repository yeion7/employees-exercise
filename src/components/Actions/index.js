import React from 'react'
import PropTypes from 'prop-types'

const Actions = ({onAddEmploy, onEdit, changeCurrency, currency, printTable}) => {
  return (
    <div>
      <button>
        Agregar empleado
      </button>
      <button>
        Editar
      </button>
      <button onClick={changeCurrency}>
        Mostrar en {currency === 'MXN' ? 'Dolares' : 'Pesos'}
      </button>
      <button onClick={printTable}>
        Imprimir en consola
      </button>
    </div>
  )
}

Actions.propTypes = {
  onAddEmploy: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
  changeCurrency: PropTypes.func.isRequired
}

export default Actions
