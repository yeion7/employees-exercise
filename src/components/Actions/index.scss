.action {
  &__container {
    margin: 15px 0;
  }

  &__button {
    &--primary {
      background: transparent;
      border-color: mediumseagreen;
      padding: 10px;
      margin: 5px;
      border-radius: 4px;
      font-size: 1em;
      transition: all 0.4s ease 0s;
      min-height: 50px;

      &:hover {
        background: mediumseagreen;
        color: white;
      }
    }
  }
}
