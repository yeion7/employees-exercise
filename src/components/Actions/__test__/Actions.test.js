import React from 'react'
import renderer from 'react-test-renderer'

import Actions from '../'

describe('Actions component', () => {
  it('renders correctly with MXN and not editable', () => {
    const fn = jest.fn()

    const tree = renderer
      .create(
        <Actions
          toggleEdit={fn}
          changeCurrency={fn}
          currency="MXN"
          printTable={fn}
          editable={false}
          addEmploy={fn}
          returnOriginal={fn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with USD and  editable', () => {
    const fn = jest.fn()

    const tree = renderer
      .create(
        <Actions
          toggleEdit={fn}
          changeCurrency={fn}
          currency="USD"
          printTable={fn}
          editable={true}
          addEmploy={fn}
          returnOriginal={fn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
