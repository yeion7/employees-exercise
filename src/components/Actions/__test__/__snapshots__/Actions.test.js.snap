// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions component renders correctly with MXN and not editable 1`] = `
<div
  className="action__container"
>
  <button
    aria-label="Agregar nuevo empleado"
    className="action__button--primary"
    data-test="add"
    onClick={[Function]}
  >
    <span
      aria-label="agregar empleado con emoji sumar"
      role="img"
    >
      ➕ Agregar empleado
    </span>
  </button>
  <button
    aria-label="Editar"
    className="action__button--primary"
    data-test="edit"
    onClick={[Function]}
  >
    <span
      aria-label="editar con emoji escribir"
      role="img"
    >
      📝 Editar
    </span>
  </button>
  <button
    aria-label="Mostrar salario en Dolares"
    className="action__button--primary"
    data-test="currency"
    onClick={[Function]}
  >
    <span
      aria-label="mostrar en Dolares con emoji moneda"
      role="img"
    >
      💰 Mostrar en 
      Dolares
    </span>
  </button>
  <button
    aria-label="Imprimir en consola"
    className="action__button--primary"
    data-test="print"
    onClick={[Function]}
  >
    <span
      aria-label="imprimir en consola con emoji imprimir"
      role="img"
    >
      🖨 Imprimir en consola
    </span>
  </button>
  <button
    aria-label="Eliminar cambios"
    className="action__button--primary"
    data-test="delete"
    onClick={[Function]}
  >
    <span
      aria-label="Eliminar cambios con emoji basura"
      role="img"
    >
      🗑 Eliminar cambios
    </span>
  </button>
</div>
`;

exports[`Actions component renders correctly with USD and  editable 1`] = `
<div
  className="action__container"
>
  <button
    aria-label="Agregar nuevo empleado"
    className="action__button--primary"
    data-test="add"
    onClick={[Function]}
  >
    <span
      aria-label="agregar empleado con emoji sumar"
      role="img"
    >
      ➕ Agregar empleado
    </span>
  </button>
  <button
    aria-label="Guardar cambios"
    className="action__button--primary"
    data-test="edit"
    onClick={[Function]}
  >
    <span
      aria-label="guardar cambios con emoji guardar"
      role="img"
    >
      💾 Guardar
    </span>
  </button>
  <button
    aria-label="Mostrar salario en Pesos"
    className="action__button--primary"
    data-test="currency"
    onClick={[Function]}
  >
    <span
      aria-label="mostrar en Pesos con emoji moneda"
      role="img"
    >
      💰 Mostrar en 
      Pesos
    </span>
  </button>
  <button
    aria-label="Imprimir en consola"
    className="action__button--primary"
    data-test="print"
    onClick={[Function]}
  >
    <span
      aria-label="imprimir en consola con emoji imprimir"
      role="img"
    >
      🖨 Imprimir en consola
    </span>
  </button>
  <button
    aria-label="Eliminar cambios"
    className="action__button--primary"
    data-test="delete"
    onClick={[Function]}
  >
    <span
      aria-label="Eliminar cambios con emoji basura"
      role="img"
    >
      🗑 Eliminar cambios
    </span>
  </button>
</div>
`;
